### ISSUE TITLE
Registration validation

### ISSUE THREAD

**Issue by zdebel** [usememos/memos#3994]
### Describe the solution you'd like

It would be great if I could manually accept user registrations. When I have for example SSO turned on, anyone can register and use my memos deployment. I'd like to be able to accept who gets access and who doesn't.

### Type of feature

User Experience (UX)

**Comment by MehadND** [comment]
So if I understand it correctly you want users to register but restrict their access until you, as the admin, manually review and accept their request and only then they should be given full access.

**Comment by zdebel** [comment]
Yes, exactly. The scenario is I have a memos instance for family and friends, it's easy for them to use Gmail as an auth option, but random people can also register, that doesn't suite me. If I can review who gets in, it makes it great.

**Comment by MehadND** [comment]
yeah I never worked on anything like this before but I will try to look into it

**Comment by RoccoSmit** [comment]
This seems like a sizable feature, regardless of if it gets built, if you arent expecting too many people to join you could turn on registration in the settings till your family/friends registered and then turn off registrations after. Or you can turn off registration and send out customer logins

**Comment by whiletoken** [comment]
> This seems like a sizable feature, regardless of if it gets built, if you arent expecting too many people to join you could turn on registration in the settings till your family/friends registered and then turn off registrations after. Or you can turn off registration and send out manually created logins

but I noticed that the latest version removes the environment variable about public for whether to enable registration or not.

**Comment by whiletoken** [comment]
1167df29d8dbb17ec54f3b941c6851faccc06bb6  Command for public was deleted, So, How to enable or disable registration in the current version

**Comment by RoccoSmit** [comment]
That removal is from container startup, its a setting now so you can turn it on and off as you require
![image](https://github.com/user-attachments/assets/0de449ee-520f-4e77-941d-cf1237b7d064)

**Comment by whiletoken** [comment]
there is no system settting config on my google chrome, system os is debian and macos catalina, but firefox is working fine

**Comment by RoccoSmit** [comment]
Only `host` level users get to see the `system` menu item in `settings`

**Comment by whiletoken** [comment]
![image](https://github.com/user-attachments/assets/47ff3e3e-47a8-47d8-814f-681d102aa8cf)
![image](https://github.com/user-attachments/assets/243eeaec-ac47-460c-823f-2472ecbedeff)

**Comment by whiletoken** [comment]
> ![image](https://private-user-images.githubusercontent.com/13046147/376010494-47ff3e3e-47a8-47d8-814f-681d102aa8cf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg3OTg5MTYsIm5iZiI6MTcyODc5ODYxNiwicGF0aCI6Ii8xMzA0NjE0Ny8zNzYwMTA0OTQtNDdmZjNlM2UtNDdhOC00N2Q4LTgxNGYtNjgxZDEwMmFhOGNmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEzVDA1NTAxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRmMjJkMjFhNTQ5ZWJiMzNlN2FhOGM2MDFhMzZjZGIzYTE3N2M2ZWYxZjdiZTI1YjIyZmU0YzUyY2I0MjU4OGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.IovQyMCay0xMcYBw9VzHkcrCnZM17FAQIlNBTb37CU0) ![image](https://private-user-images.githubusercontent.com/13046147/376010589-243eeaec-ac47-460c-823f-2472ecbedeff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg3OTg5MTYsIm5iZiI6MTcyODc5ODYxNiwicGF0aCI6Ii8xMzA0NjE0Ny8zNzYwMTA1ODktMjQzZWVhZWMtYWM0Ny00NjBjLTgyM2YtMjQ3MmVjYmVkZWZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEzVDA1NTAxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzZWM5YTMwNTlmOGI0M2E5NTc4MWRlOGFjZjliMTZjOTVkZWNjMjJkMjJhMGU2YWFiNzM4NmM3NzFkYzgwNDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.CsdGILi6VdfutUQSs-6tE8p1WWtOJWydfksyU4BNVn4)

the first is google chrome, the second is firefox

**Comment by whiletoken** [comment]
google chrome version is  129.0.6668.100

**Comment by RoccoSmit** [comment]
Unable to replicate. Do you have any browser extensions in chrome that would remove the content?

**Comment by whiletoken** [comment]
> Unable to replicate. Do you have any browser extensions in chrome that would remove the content? Or perhaps some customer javascript in your `Additional script` section

thx, you are right,working fine on chrome's incognito mode

**Comment by zdebel** [comment]
@johnnyjoygh this seems like one way to achieve what I want, not very user friendly, but it should work, thanks for the tip!

**Comment by ElTopo** [comment]
I also host a memos site for family members and friends only. I have this workaround:

-  As host user, in Settings/System: enable "Disallow user registration", disable "Disallow password auth"
-  In Settings/Member, create a new user upon request, with an initial password
-  Ask the new user to log in and change their password; if they want, they can also change their "Username" (disable "Disallow Change Username") and "Nickname" (disable "Disallow Change Nickname")
